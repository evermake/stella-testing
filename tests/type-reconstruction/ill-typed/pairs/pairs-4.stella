language core;

extend with #pairs, #type-reconstruction;

fn main(x : auto) -> Nat {
  return
    (fn(x : auto) { return { { { x, x }, { x, x } }, { { x, Nat::iszero(x) }, { x, x } } } })(0).2.1
}
