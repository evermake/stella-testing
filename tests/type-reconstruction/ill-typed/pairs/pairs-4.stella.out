An error occurred during typechecking!
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
when unifying expected type
  Nat
against actual type
  
{
  Nat, Bool
}

when typechecking the expression
  (fn (x : ?T2)
{
  return
  {
    {
      {
        x, x
      }
      ,
      {
        x, x
      }
    }
    ,
    {
      {
        x, Nat::iszero (x)
      }
      ,
      {
        x, x
      }
    }
  }
}
) (0) . 2 . 1